src/FormContainer.tsx

```diff
@@ -98,20 +105,34 @@ const makeWrapper = <T extends {}>(config: IFormConfig<T>) => (WrappedComponent:
             }
         });
 
+        handleSubmit = (submit: () => Promise<any>) => () => {
+            return submit().catch((error: any) => {
```

Comment:
or values can be added automatically upon the invocation:
`return submit(this.state.model).catch(...`

you need something to validate, right?

Reply:
Good question. In our usage we have access to the form model in props for the container where our `submit` function is defined so we simply destructure props inside that function e.g.

```
submit = () => {
    const {
      form: { model },
    } = this.props;

    ...
}
```

Injecting them via the handleSubmit is a good idea, though, as `submit` could potentially be a static function then! You happy for me to just pass the model? Do you think the submit function should have access to anything else?

Reply:
Updated to pass `model` for now!